docs/demos/Smooth_Platform_Coordinate_Transitions.py

```diff
@@ -0,0 +1,214 @@
+
+fig = plt.figure(figsize=(10, 6))
+ax = fig.add_subplot(1, 1, 1)
+ax.set_xlabel("$x$")
+ax.set_ylabel("$y$")
+ax.axis('equal')
+ax.scatter(X, Y, color='lightskyblue', s=100, edgecolors='black')
+ax.plot((X[0], X[0] + 2*ux), (Y[0], Y[0] + 2*uy), color='lightgreen', linewidth=2)
+
+
+# %%
+# :meth:`~simulator.transition.get_smooth_transition_models` requires the initial state of the
+# platform, x and y coordinates, times to be at each respective coordinate and the platform
+# turn-rate.
+#
+# The 'times' parameter will dictate the times at which the target should be at each respective
+# target coordinate. Hence, the first coordinate transition should take 10 seconds.
+# The target starts off with insufficient speed, so we expect it to accelerate once out of the turn
+# so as to get to the destination in a total of 10 seconds.
+#
+# The target will have a constant turn rate of 25 degrees per second - a sharp turn, partly to
+# guarantee that it can reach its destination in time, but also as initial velocity and angle
+# turn-rate are constrained according to the following formulae (derived for the case where
+# turn-rate (:math:`w`) and initial bearing (:math:`a`) are positive and :math:`< 180` degrees):
+#
+# .. image:: ../_static/deriving_constant_turn_angle.png
+#   :width: 900
+#   :alt: Derivation of constant turn equation
+#
+# where we must have that :math:`c \leq \frac{d\sin(b)}{\sin(wt)}`. Ie. the turn must be completed
+# without the target 'over-shooting' the destination.
+# Along with the constraint that the turn must be possible - if the turn-rate is too low, the
+# target may never turn enough to hit the destination.
+
+from stonesoup.simulator.transition import create_smooth_transition_models
+transition_models, transition_times = create_smooth_transition_models(initial_state=platform_state,
+                                                                      x_coords=X,
+                                                                      y_coords=Y,
+                                                                      times=times,
+                                                                      turn_rate=np.radians(25))
+
+# %%
+# This gives the transition times/models:
+from stonesoup.models.transition.linear import ConstantTurn
+for transition_time, transition_model in zip(transition_times, transition_models):
+    if not isinstance(transition_model, ConstantTurn):
+        if transition_model.over_shot:
+            model = transition_model.new_model
+    print('Duration: ', transition_time.total_seconds(), 's ',
+          'Model: ', type(transition_model), end=' ')
+    if isinstance(transition_model, ConstantTurn):
+        print('turn-rate: ', transition_model.turn_rate)
+    else:
+        print('x-acceleration: ', transition_model.ax, end=', ')
+        print('y-acceleration: ', transition_model.ay)
+
+# %%
+# The last transition model calls a :class:`~.Point2PointStop` transition model to bring the target
```

Comment:
This overshoot: It's not clear to me why a linear transition would lead to an overshoot. Surely it could just accelerate less. Maybe this could be clarified?

Reply:
In a situation where linearly decelerating to zero speed exactly at the next coordinate leads to the platform arriving there early, there is an issue with overshooting. If we were to apply a stronger deceleration, the platform would hit zero speed before the next coord. So a weaker deceleration is needed, hence the platform will go past the next coord and reverse back.

Reply:
Ahh of course.